examples/fairmotion_interface_utils.py

```diff
@@ -0,0 +1,342 @@
+    ## SCENIC ##
+
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_08_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    drink_beverage = MotionData(motion_, MType.SCENIC, transitive_motion_=walk_to_walk)
+
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_22_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    wash_window = MotionData(
+        motion_ops.cut(motion_, 3040, 4090),
+        MType.SCENIC,
+        transitive_motion_=walk_to_walk,
+    )
+
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_23_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    sweep_floor = MotionData(motion_, MType.SCENIC, transitive_motion_=walk_to_walk)
+
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_29_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    jumping_jacks = MotionData(
+        motion_, MType.SCENIC, transitive_motion_=run_to_run, scenic_direction_offset=90
+    )
+
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_10_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    reach_for = MotionData(
+        motion_,
+        MType.SCENIC,
+        transitive_motion_=walk_to_walk,
+        scenic_direction_offset=-80,
+    )
+
+
+class PathData:
+    """
+    The PathData class is purposed to instantiate with given path points, and used
+    to manage the path data fro the current path following sequence.
+    """
+
+    def __init__(self, path_points: List[mn.Vector3]) -> None:
+        self.points = path_points
+        self.length = self.calc_path_length(path_points)
+        self.time = 0.0
+
+    # [REFACTOR] I would like to change this to a static method, once I figure out how
+    def calc_path_length(self, path_points: List[mn.Vector3]) -> float:
+        # get path length
+        i, j, summ = 0, 0, 0.0
+        while i < len(path_points):
+            summ += (mn.Vector3(path_points[i] - path_points[j])).length()
+            j = i
+            i += 1
+        return summ
+
+    def __str__(self) -> str:
+        return f"PathData(points={type(self.points)}, length={self.length}, time={self.time})"
+
+
+@dataclass
+class ActionOrder:
+    """
+    The ActionOrder class holds the data necessary to command the pathfollower character
+    to perform a scenic motion at a specified location.
+    """
+
+    def __init__(
+        self,
+        motion_data: Motions.MotionData,
+        location: Optional[mn.Vector3] = None,
+        facing: Optional[mn.Vector3] = None,
+    ) -> None:
+        self.motion_data = motion_data
+
+        self.location = None
+        if location is not None:
+            self.location = mn.Vector3(location)
```

Comment:
Cast/checking should not be necessary if typing (Optional[Vector3]) is correct.
```suggestion
        self.location = location
```
Same with `self.facing` below